python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -0,0 +1,1111 @@
+                    self._reference_array[ref_id], ref_id, ref_delta,
+                    ref_dist, tmp_ref_delta_array, tmp_ref_dist_arary,
+                    tmp_ref_id_array, max_dist_rad,
+                    n_match)
+
+                # Get the data from the return struct.
+                ref_spoke_list = pattern_spoke_struct.ref_spoke_list
+                src_spoke_list = pattern_spoke_struct.src_spoke_list
+                # If we don't find enough candidates we can continue to the
+                # next reference center pair.
+                if len(ref_spoke_list) < n_match - 2 or \
+                   len(src_spoke_list) < n_match - 2:
+                    continue
+
+                # If we have the right number of matched ids we store these.
+                matched_ref_list.extend(ref_spoke_list)
+                matched_src_list.extend(src_spoke_list)
+
+                # We can now create our full rotation matrix for both the
+                # shift and rotation. Reminder shift, alings the pattern
+                # centers, rotation rotates the spokes on top of each other.
+                shift_rot_struct = self._create_shift_rot_matrix(
+                    cos_rot_sq, shift_matrix, src_delta_array[0],
+                    self._reference_array[ref_id], ref_delta)
+                # If we fail to create the rotation matrix, continue to the
+                # next objects.
+                if shift_rot_struct is None:
+                    continue
+
+                # Get the data from the return struct.
+                sin_rot = shift_rot_struct.sin_rot
+                shift_rot_matrix = shift_rot_struct.shift_rot_matrix
+
+                # Now that we have enough canidates we test to see if it passes
+                # intermediate verify. This shifts and rotates the soruce
+                # pattern into the reference frame and tests that each
+                # source/reference object pair is within max_dist.
+                if self._intermediate_verify(src_pattern_array[
+                                                 matched_src_list],
+                                             self._reference_array[
+                                                 matched_ref_list],
+                                             shift_rot_matrix, max_dist_rad):
+                    return pipeBase.Struct(
+                        ref_candidates=matched_ref_list,
+                        src_candidates=matched_src_list,
+                        shift_rot_matrix=shift_rot_matrix,
+                        cos_shift=cos_shift,
+                        sin_rot=sin_rot)
+        return None
+
+    def _find_candidate_reference_pairs(self, src_dist, ref_dist_array,
+                                        max_dist_rad):
+        r"""Wrap numpy.searchsorted to find the range of reference spokes within
+        a spoke distance tolernace of our source spoke. Returns an array sorted
```

Comment:
Brief description followed by blank line followed by details. (Here and in all doc strings)

"tolernace" -> "tolerance"

Returns -> Return (active voice) (here and in all doc strings)